### PR TITLE
Big zenbuild clean-up as part of ARM work

### PR DESCRIPTION
**Pull request by rvs**

### CODE REVIEW

**Comment by rvs** [comment]
@gianlzed fair enough. In fact, I'm actually trying to do it right now. The problem is -- I can't make them totally independent, so its not like you'll be able to cherry-pick randomly (although some you would). That said, at this point it is definitely not a request to merge, but rather to get your feedback on the overall direction of where this is going. So in that sense -- the cumulative changes are more relevant, no?

**Comment by rvs** [comment]
@gianlzed @eriknordmark and now I present 9 separate commits for your [re-]viewing pleasure. All nicely separated and [mostly] self-contained.

**Comment by rvs** [comment]
@eriknordmark interestingly enough running Xen under ARM/qemu feels waaaaay more stable  than doing the same under x86. Time drift hasn't been a problem so far and it also feels much snappier than x86. To a point where I may actually prefer testing with that combination even on my x86 laptop.

**Comment by gianlzed** [comment]
To me this seems mostly changes due to personal style. I am okay with it, although I think a few things make stuff more hard to follow. As long as @rvs and @eriknordmark are okay with it I don't see why you don't want to push this thing in.

**Comment by rvs** [comment]
@gianlzed agree on both accounts. Right now the exercise has been to minimize the # of things that need to work on both ARM and x86 (hence me ditching a dedicated installer package and just re-using the mkflash container, etc.). The idea is that it will be easier for us to get to a state where ARM is on par with x86 and then we can grown back from that point on while keeping both ARM and x86 happy.

As for the person style -- yes ;-) Now that I'm basically assuming the maitainership of this thing -- I figured I might as well ;-)
